plugins/native_dio_adapter/lib/src/cronet_adapter.dart

```diff
@@ -11,16 +14,32 @@ class CronetAdapter implements HttpClientAdapter {
   CronetAdapter(
     CronetEngine? engine, {
     bool closeEngine = true,
-  }) : _conversionLayer = ConversionLayerAdapter(
-          engine == null
-              ? CronetClient.defaultCronetEngine()
-              : CronetClient.fromCronetEngine(engine, closeEngine: closeEngine),
-        );
+  })  : _engine = engine,
+        _closeEngine = closeEngine;
 
-  final ConversionLayerAdapter _conversionLayer;
+  final CronetEngine? _engine;
+  final bool _closeEngine;
 
-  /// The underlying conversion layer adapter.
-  ConversionLayerAdapter get adapter => _conversionLayer;
+  late final ConversionLayerAdapter _conversionLayer = () {
+    Client client;
+
+    try {
+      client = CronetClient.fromCronetEngine(
+        _engine ?? CronetEngine.build(),
+        closeEngine: _closeEngine,
+      );
+    } catch (error, stackTrace) {
```

Comment:
This is too broad that includes unrelated errors. It must be the precise one with the exact type and message.